btrfs.h

```diff
@@ -15,22 +15,28 @@
 */
 
 #include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+#include <linux/version.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,7,0)
```

Comment:
Good idea! I'd verify that this is true for containerd however; the `btrfs_tree.h` header was introduced in kernel 4.7, but I'm not 100% sure that all the function containerd uses was present in that version without verification.

Reply:
(For a simple check, spin up a disposable system, fetch a tarball of kernel v4.7, and `rm -rf /usr/include/linux && make headers_install` while in the source tree. Build and see if things explode.)

Reply:
Looks like `make headers_install` does not install `btrfs_tree.h` with kernel 4.7 👀

Reply:
Odd... It's definitely present in the v4.7 source tree (https://github.com/torvalds/linux/blob/v4.7/include/uapi/linux/btrfs_tree.h), and I am able to compile Moby with my patches just fine by using a kernel source checkout at v4.7 and running `make headers_install` (no kernel headers were present in the container before the `make` invocation).

Reply:
The minimal requirement turned out to be kernel 4.12
https://github.com/torvalds/linux/commit/fcc8487d477a3452a1d0ccbdd4c5e0e1e3cb8bed

Reply:
> Odd... It's definitely present in the v4.7 source tree (https://github.com/torvalds/linux/blob/v4.7/include/uapi/linux/btrfs_tree.h), and I am able to compile Moby with my patches just fine by using a kernel source checkout at v4.7 and running `make headers_install` (no kernel headers were present in the container before the `make` invocation).

Could you verify https://github.com/moby/moby/pull/44762 ?

Reply:
It looks like you found the right commit -- and it looks like I had my source tree at v4.14 and forgot to verify with `git describe` -- oops!